staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	genericvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+)
+
+func ValidateInitializerConfiguration(ic *apiserver.InitializerConfiguration) field.ErrorList {
+	allErrors := genericvalidation.ValidateObjectMeta(&ic.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	for i, initializer := range ic.Initializers {
+		allErrors = append(allErrors, validateInitializer(&initializer, field.NewPath("initializers").Index(i))...)
+	}
+	return allErrors
+}
+
+func validateInitializer(initializer *apiserver.Initializer, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	// TODO: update IsQualifiedName
```

Comment:
@deads2k the code history is lost during the move to staging. It looks like we used to have a completely different criteria on [QualifiedName](https://github.com/kubernetes/kubernetes/blob/5b3d0bbe66343e14eade6b1c746f868aee1a6fa9/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L41). Shall I create my own function?